qiita_db/handlers/plugin.py

```diff
@@ -0,0 +1,186 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from json import loads, dumps
+
+from tornado.web import HTTPError
+
+from .oauth2 import OauthBaseHandler, authenticate_oauth
+import qiita_db as qdb
+
+
+def _get_plugin(name, version):
+    """Returns the plugin with the given name and version
+
+    Parameters
+    ----------
+    name : str
+        The name of the plugin
+    version : str
+        The version of the plugin
+
+    Returns
+    -------
+    qiita_db.software.Software
+        The requested plugin
+
+    Raises
+    ------
+    HTTPError
+        If the plugin does not exist, with error code 404
+        If there is a problem instantiating the plugin, with error code 500
+    """
+    try:
+        plugin = qdb.software.Software.from_name_and_version(name, version)
+    except qdb.exceptions.QiitaDBUnknownIDError:
+        raise HTTPError(404)
+    except Exception as e:
```

Comment:
I'm unsure on how to force those errors to happen. I put those as a security guard in case that something unexpected happens (that's why they catch any exception). Other endpoints are similar to this.
Any idea on how to force the error?